src/CourseById.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { Course } from "./Courses";
+import { Box, Button, Card, TextField, Typography } from "@mui/material";
+
+export default function CourseById() {
+  const [course, setCourse] = useState([]);
```

Comment:
Use null, here as empty [] is truthy value, !course will not execute anytime.

Reply:
changed to null